dsl/doc.go

```diff
@@ -77,5 +77,74 @@ The example above can thus be simplified to:
         })
     })
 
+The GRPC DSL adds a "GRPC" function to the DSL constructs that require gRPC
+specific information. These include the API, Service, Method, and Error DSLs.
+
+For example:
+
+    var _ = API("name", func() {
+        Description("Optional description")
+        // gRPC specific properties
+        GRPC(func() {
+        })
+    })
+
+The GRPC function defines the mapping of the data type attributes used in the
+generic DSL to gRPC messages and metadata.
+
+For example:
+
+    var PayloadType = Type("Payload", func() {
+        TypeName("Payload")        // mapped to gRPC message with name "Payload"
+        Field(1, "rq1", String)    // mapped to field in "Payload" message
+        													 // with name "rq1" and tag number 1
+        Field(2, "rq2", String)    // mapped to field in "Payload" message
+        													 // with name "rq2" and tag number 2
+        Attribute("rq3", Int)
+        Attribute("rq4", Int)
+    })
+
+    var ResultType = ResultType("application/vnd.result", func() {
+        TypeName("Result")         // mapped to gRPC message with name "Result"
+        Attributes(func() {
+           Attribute("rp1", Int)
+           Field(1, "rp2", String) // mapped to field in "Result" message
+        													 // with name "rp2" and tag number 1
+        })
+    })
+
+    var _ = Service("name", func() {
+    		Method("name", func() {
+           Payload(PayloadType)
+           Result(ResultType)       // has attributes rp1 and rp2
+           Error("name")
+
+    			 GRPC(func() {
+               Metadata(func() {   // rq3 and rq4 present in gRPC request metadata
+                   Attribute("rq3)
+                   Attribute("rq4")
+               })
+               Message(func() {
+                   Attribute("rq1") // rq1 and rq2 present in gRPC request message
+               })
+               Response(CodeOK, func() {
+                   Metadata(func() {
+                       Attribute("rp1") // rp1 present in gRPC response metadata
+                   })
+                   Message(func() {
+                       Attribute("rp2") // rp2 present in gRPC response message
+                   })
+               })
+               Response("name", CodeInternal)
```

Comment:
Might good to document that this response represents the response message used in case of error for error "name". I forget what we decided about encoding error structs?

Reply:
Right now, gRPC sends a default error message of this type - https://godoc.org/google.golang.org/grpc/status#Status . We don't really support custom error type yet.

Reply:
So we have a error type to represent that in the design? I seem to recall we said we should simplify the HTTP error type to match that.